Add support for color names

ghid opened the pull request:
Hi. 
First of all: A very great Plugin! Using the SignColumn to visualize the colors is awsome.

I've added a pattern to match also HTML color names. I'd feel glad if you'd include it to your project.

Some examples:
![image](https://user-images.githubusercontent.com/7606219/55879563-3514e400-5b9f-11e9-82e4-e1fea2173c68.png)

Thanks for your time and best wishes
Peter

Commit:
Add support for color names

RRethy commented on autoload/hexokinase/patterns/color_names.vim:
```diff
@@ -0,0 +1,155 @@
+	  \ }
+
+function! hexokinase#patterns#color_names#get_pattern() abort
+  return '\c\<\(' . join(keys(s:COLORS), '\|') . '\)\>'
```
Having `\c` here means the rest of the pattern will be case insensitive. It should either be removed or the case-sensitivity of the pattern should be restored.

RRethy commented on autoload/hexokinase/patterns/color_names.vim:
```diff
@@ -0,0 +1,155 @@
+let s:COLORS = {
```
```suggestion
let s:colours = {
```

RRethy replied:
Can the indentation for the colors be changed from 1tab+2spaces to 6 spaces (before the `\`).

RRethy reviewed (changes requested):
I am hesitant with this change. I had these colours in briefly but I removed them because of the size of the regex and the possible performance impact. I'm going to do some benchmarking later this week on this PR to see the size of the performance hit with large files that contain long lines.

Another option that I would willing to merge without any benchmarking would be to make patterns opt-in. Something along the lines of `g:Hexokinase_optInPatterns = ['6hex', '3hex', 'rgb', 'rgba']` (this would be default) then 'names' would be something the user could opt into. This could pave the way to add more patterns without hurting performance since currently large files with long lines are not the most performant which is not something I want to make worse.

RRethy reviewed (changes requested):
The following colors are missing (https://www.w3schools.com/colors/colors_names.asp)

```
'darkgrey': '#a9a9a9'
'darkslategrey': '#2f4f4f'
'dimgrey': '#696969'
'grey': '#808080'
'lightgrey': '#d3d3d3'
'lightslategrey': '#778899'
'rebeccapurple': '#663399'
'slategrey': '#708090'
```

And `mediumpurple` should have the hex code `#9370db` rather than `#9370d8`.

RRethy commented:
First off, thanks for the PR, hopefully my extensive scrutinizing doesn't come off too negatively, I do appreciate people putting in the time to improve the plugin 😄 

As well, I just did some benchmarking, unfortunately this has a significant impact on the amount of time it takes to do parse the file and match colours so I won't have this on by default. I'm going to push a change to master which allows the user to customize which patterns are used, then you can rebase and make the colour names an opt-in pattern.

RRethy commented on autoload/hexokinase/patterns/color_names.vim:
```diff
@@ -0,0 +1,155 @@
+function! hexokinase#patterns#color_names#process(str) abort
+  let str_lcase = tolower(a:str)
+  if has_key(s:COLORS, str_lcase)
+	return s:COLORS[str_lcase]
```
Change tab to 4 spaces.

RRethy commented on autoload/hexokinase/patterns/color_names.vim:
```diff
@@ -0,0 +1,155 @@
+  if has_key(s:COLORS, str_lcase)
+	return s:COLORS[str_lcase]
+  else
+	return ''
```
Change tab to 4 spaces.

Commit:
Perform requested changes

- Added missing colours
- Correct wrong colour codes
- Change case of script var to lowercase
- Change indentation to 4 spaces instead of tabs

RRethy commented:
@ghid , [this](https://github.com/RRethy/vim-hexokinase/commit/3cc556d0fb5a3cb4ef1205f2a78262d264dc5865) commit will let this be an opt in feature, please add it under the name `colour_names`.

As well, please indent with 2 spaces. And 6 spaces before the `\` line continuation.

Commit:
Do not add colour-name pattern by default

Commit:
Rename file color_names.vim to colour_names.vim

ghid commented:
> @ghid , [this](https://github.com/RRethy/vim-hexokinase/commit/3cc556d0fb5a3cb4ef1205f2a78262d264dc5865) commit will let this be an opt in feature, please add it under the name `colour_names`.
> 
> As well, please indent with 2 spaces. And 6 spaces before the `\` line continuation.

@RRethy , I applied your change requests. Would you take a look, please?
